SPECS/python-conda-package-streaming/python-conda-package-streaming.spec

```diff
@@ -0,0 +1,143 @@
+%global srcname conda-package-streaming
+%global pkgname conda_package_streaming
+
+# We have a circular dep on conda for tests
+%bcond_with bootstrap
+
+Distribution:   Azure Linux
+Name:           python-%{srcname}
+Vendor:         Microsoft Corporation
+Version:        0.11.0
+Release:        2%{?dist}
+Summary:        Extract metadata from remote conda packages without downloading whole file
+
+License:        BSD-3-Clause
+URL:            https://github.com/conda/conda-package-streaming
+Source0:        https://github.com/conda/%{srcname}/archive/v%{version}/%{srcname}-%{version}.tar.gz
+
+BuildArch:      noarch
+
+%global common_description %{expand:Download conda metadata from packages without transferring entire file. Get
+metadata from local .tar.bz2 packages without reading entire files.
+
+Uses enhanced pip lazy_wheel to fetch a file out of .conda with no more than
+3 range requests, but usually 2.
+
+Uses tar = tarfile.open(fileobj=...) to stream remote .tar.bz2. Closes the
+HTTP request once desired files have been seen.}
+
+%description
+%{common_description}
+
+
+%package -n python3-%{srcname}
+Summary:        %{summary}
+BuildRequires:  python3-devel
+BuildRequires:  python3-flit-core
+BuildRequires:  python3-pip
+Requires:       python3-requests
+Requires:       python3-zstandard
+# For tests
+%if %{without bootstrap}
+# Need conda executable for tests
+BuildRequires:  conda
+%endif
+
+%description -n python3-%{srcname}
+%{common_description}
+
+
+%prep
+%autosetup -n %{srcname}-%{version}
+# do not run coverage in pytest, drop unneeded and unpackaged boto3-stubs dev dep
+sed -i -e '/cov/d' -e '/boto3-stubs/d' pyproject.toml requirements.txt
+%if %{with bootstrap}
+sed -i -e '/"conda"/d' -e '/conda-package-handling/d' pyproject.toml
+%endif
+
+%generate_buildrequires
+%pyproject_buildrequires -x test
+
+%build
+%pyproject_wheel
+
+%install
+%pyproject_install
+%pyproject_save_files %{pkgname}
+
+%check
+pip3 install pytest pytest-cov pytest-mock boto3 boto3-stubs[essential] bottle zstandard archspec
```

Comment:
Whatever versions are working with this pip3 install, can we hard-code them to the specific versions so that the tests remain passing for as long as possible?